src/actions/send.js

```diff
@@ -0,0 +1,40 @@
+import { createActions } from 'redux-actions';
+
+import { showAlert } from '../utils/alerts';
+import { wallet } from '../utils/wallet';
+
+export const transfer = ({ contractName, amount, memo, receiverId, isStorageBalanceAvailable }) => async (dispatch, gesState) => {
+    if (contractName && !isStorageBalanceAvailable) {
```

Comment:
Rather than making implicit assumptions around `contractName`, we want the type of token being sent to be _explicitly defined_ like we had talked about previously

e.g.
```javascript
const TOKEN_TYPES = {
  NEAR: 'NEAR',
  NEP141: 'NEP141',
  ...
};

...

export const transfer({
  type<key of TOKEN_TYPES>,
  params: {
    contractName, // Not present when type === 'NEAR', but also _not_ the way we decide whether the send is for NEAR or something else
    amount,
    memo,
    receiverId
  }
}) {

  if(type === TOKEN_TYPES.NEAR) {
    // **All** logic for handling NEAR goes here
  } else if(type === TOKEN_TYPES.NEP141) {
    // **All** logic for handling NEP141 tokens goes here
  } else {
    throw new TypeError(`Could not transfer unsupported token: ${type}`);
  }
}
```

This will be a more flexible structure in the long run, easier to extend later without touching logic for existing implementations of tokens that we already support, and much easier to read & debug.

Reply:
Done.